index.js

```diff
@@ -78,17 +79,22 @@ export class StuckTransactionsCanceller {
 
   async removeConfirmed (tx) {
     assert.strictEqual(typeof tx.hash, 'string')
-    await this.#store.remove(tx.hash)
+    const txs = await this.#store.list()
+    for (const _tx of txs) {
+      if (_tx.nonce === tx.nonce) {
+        await this.#store.remove(_tx.hash)
+      }
+    }
```

Comment:
Can there be a race condition where another replacement transaction is added in the time window between `this.#store.list()` and `this.#store.remove(_tx.hash)`?

Reply:
I don't see an obvious one 🤔